### PR TITLE
update usage for 'docker build' with '--progress' and '--secret' options

### PR DESCRIPTION
**Pull request by AntaresS**
**- What I did**
- updated `docker build` usage with `--progress` option and `--secret` option

**- How I did it**

**- How to verify it**

**- Description for the changelog**

Signed-off-by: Anda Xu <anda.xu@docker.com>

### CODE REVIEW

**Comment by AntaresS** [comment]
cc @andrewhsu

File: docs/reference/commandline/build.md [hunk 1]
```diff
@@ -48,8 +48,11 @@ Options:
                                 '<network-name>|<network-id>': connect to a user-defined network
       --no-cache                Do not use cache when building the image
       --pull                    Always attempt to pull a newer version of the image
+      --progress                Set type of progress output (only if BuildKit enabled) (auto, plain, tty).
```
Comment (andrewhsu) [inline docs/reference/commandline/build.md]: `(only takes effect if BuildKit is enabled)`

Comment (andrewhsu) [reply docs/reference/commandline/build.md]: should probably say what the default value is

Comment (AntaresS) [reply docs/reference/commandline/build.md]: The only thing is that's according to the cli `--help` manual 
https://github.com/docker/cli/blob/master/cli/command/image/build.go#L157
I think it's better to keep them consistent.

File: docs/reference/commandline/build.md [hunk 2]
```diff
@@ -48,8 +48,11 @@ Options:
                                 '<network-name>|<network-id>': connect to a user-defined network
       --no-cache                Do not use cache when building the image
       --pull                    Always attempt to pull a newer version of the image
+      --progress                Set type of progress output (only if BuildKit enabled) (auto, plain, tty). 
+                                Use plain to show container output
   -q, --quiet                   Suppress the build output and print image ID on success
       --rm                      Remove intermediate containers after a successful build (default true)
+      --secret                  Secret file to expose to the build (only if BuildKit enabled): id=mysecret,src=/local/secret"
```
Comment (thaJeztah) [inline docs/reference/commandline/build.md]: Note that the usage section of these markdown files are only used on github; for docs.docker.com, this part is generated from code (and equivalent to the information that's shown for the `--help` output)

**Pull request merged**